### PR TITLE
Maintain send window only if ExchangeMergeSortSourceOperator

### PR DESCRIPTION
**Pull request by liuyehcf**
Continue of previous pr [pr-2051](https://github.com/StarRocks/starrocks/pull/2051) 、[pr-2170](https://github.com/StarRocks/starrocks/pull/2170)

Since normal ExchangeSourceOperator needn't keep order at receiver side, therefore, the sender does not need to maintain the sending window

### CODE REVIEW

**Review by satanson — DISMISSED** [review]

File: gensrc/thrift/DataSinks.thrift [hunk 1]
```diff
@@ -82,11 +82,24 @@ struct TDataStreamSink {
...
 
 struct TMultiCastDataStreamSink {
     1: required list<TDataStreamSink> sinks;
     2: required list< list<TPlanFragmentDestination> > destinations;
+    // Only useful in pipeline mode
+    // If receiver side is ExchangeMergeSortSourceOperator, then all the
+    // packets should be kept in order (according sequence), so the sender
+    // side need to maintain a send window in order to avoiding the receiver
+    // buffer too many out-of-order packets
+    4: optional list<bool> is_merges
```
Comment (dirtysalt) [inline gensrc/thrift/DataSinks.thrift:102]: why not reuse `is_merge` in `TDataStreamSink`?

Comment (liuyehcf) [reply gensrc/thrift/DataSinks.thrift]: right, I'll fix it

**Review by dirtysalt — DISMISSED** [review]

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Comment by satanson** [comment]
run starrocks_fe_unittest

**Review by satanson — APPROVED** [review]

**Comment by liuyehcf** [comment]
run starrocks_fe_unittest

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**